src/Orleans/Core/GrainClient.cs

```diff
@@ -381,6 +381,15 @@ public static Streams.IStreamProvider GetStreamProvider(string name)
             return RuntimeClient.Current.CurrentStreamProviderManager.GetProvider(name) as Streams.IStreamProvider;
         }
 
+        public delegate void ConnectionToClusterLostHandler(object sender, EventArgs e);
+
+        public static event ConnectionToClusterLostHandler ClusterConnectionLost;
+
+        internal static void NotifyClusterConnectionLost()
+        {
+            ClusterConnectionLost?.Invoke(null, null);
```

Comment:
If we want to pass the client when it's no longer static, and plan to do so, can we add a comment or todo noting this.  Seems like something very easy to overlook in the future.